### DISCUSSION TITLE
A noooob developer asking a noob question

### DISCUSSION THREAD

**Discussion by Asharp19** [smartcontractkit/full-blockchain-solidity-course-py#487] [Q&A]
```Solidity
contract SimpleStorage {
uint256 public favNumber;

function store(uint256 _favNumber) public {
   favNumber=_favNumber;
   }
function retrieve() public returns(uint256) {
   return favNumber;
```
          
          
 THIS code right here gives out an error in line 5th but when i put *public view* instead of public error's gone, can't understand why can someone explain please?

**Comment by cromewar** [comment]
Hello @Asharp19 the function store is not actually doing a transaction on the blockchain so they should be marked as view, please refer to [this](https://www.tutorialspoint.com/solidity/solidity_view_functions.htm) link for more information about view functions.
